mlflow/models/evaluation/base.py

```diff
@@ -0,0 +1,375 @@
+from typing import Dict, Union
+import mlflow
+import hashlib
+import time
+import numpy as np
+import pandas as pd
+import pickle
+import json
+import os
+from mlflow.exceptions import MlflowException
+from mlflow.utils.file_utils import TempDir
+from mlflow.entities import Metric, RunTag
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils import _get_fully_qualified_class_name, load_class
+from mlflow.pyfunc import PyFuncModel
+from mlflow.models.evaluation.evaluator_registry import ModelEvaluatorRegistry
+
+
+class EvaluationMetrics(dict):
+    pass
+
+
+class EvaluationArtifact:
+    def __init__(self, location, content=None):
+        self._content = content
+        self._location = location
+
+    @classmethod
+    def load_content_from_file(cls, local_artifact_path):
+        raise NotImplementedError()
+
+    @classmethod
+    def save_content_to_file(cls, content, output_artifact_path):
+        raise NotImplementedError()
```

Comment:
```suggestion
    @classmethod
    def _load_content_from_file(cls, local_artifact_path):
        raise NotImplementedError()

    @classmethod
    def _save_content_to_file(cls, content, output_artifact_path):
        raise NotImplementedError()
```

Reply:
Can we make these instance methods in case developers want to reference other instance attributes when deciding how to save content?

Reply:
change `_save_content_to_file` to be instance method. But I prefer `_load_content_from_file` to be class method, then we loading EvaluationResults we can load content via class, which is easier to use